packages/vhd-cli/src/commands/compare.js

```diff
@@ -0,0 +1,111 @@
+import { VhdFile, VhdDirectory } from 'vhd-lib'
+
+import { getSyncedHandler } from '@xen-orchestra/fs'
+import { resolve } from 'path'
+import Disposable from 'promise-toolbox/Disposable'
+
+const compareObj = (src, dest, path) => {
+  for (const key in src) {
+    if (!Object.prototype.hasOwnProperty.call(src, key)) {
+      continue
+    }
+    let same = false
+    if (typeof src[key] !== typeof dest[key]) {
+      throw new Error(
+        `Error checking header : key ${path + '/' + key} is of type *${src[key]}* in source and *${dest[key]}* in dest`
+      )
+    }
+
+    if (typeof src[key] === 'object') {
+      compareObj(src[key], dest[key], path + '/' + key)
+      continue
+    }
+
+    if (Buffer.isBuffer(src[key]) && Buffer.isBuffer(dest[key])) {
+      same = src[key].equals(dest[key])
+    } else {
+      same = src[key] === dest[key]
+    }
+    if (!same) {
+      throw new Error(
+        `Error checking header : key ${path + '/' + key} is *${src[key]}* in source and *${dest[key]}* in dest`
+      )
+    }
+  }
+}
+
+export default async args => {
+  if (args.length < 2 || args.some(_ => _ === '-h' || _ === '--help')) {
+    return `Usage: compare <source VHD> <destination> `
+  }
+  const [sourcePath, destPath] = args
+
+  await Disposable.use(async function* () {
+    const handler = yield getSyncedHandler({ url: 'file:///' })
+    let src
+
+    try {
+      src = yield VhdFile.open(handler, resolve(sourcePath))
+      // openning a file for reading doew not trigger EISDIR as long as we don't really read from it
+      await src.readHeaderAndFooter()
+    } catch (e) {
+      if (e.code === 'EISDIR') {
+        src = yield VhdDirectory.open(handler, resolve(sourcePath))
+        await src.readHeaderAndFooter()
+      } else {
+        throw e
+      }
+    }
```

Comment:
Move logic in its own function (in this file for now, we'll move it to `Vhd` later).